src/Files.App/Views/Shells/ModernShellPage.xaml

```diff
@@ -99,9 +107,40 @@
 				Symbol="Forward" />
 		</Border>
 
+		<!--  Filter header  -->
+		<Grid
+			x:Name="FilterHeader"
+			Grid.Row="0"
+			Padding="26,8,12,8"
+			BorderBrush="{ThemeResource DividerStrokeColorDefaultBrush}"
+			BorderThickness="0,0,0,1"
+			Visibility="{x:Bind ShellViewModel.ShowFilterHeader, Mode=OneWay}">
+			<StackPanel
+				VerticalAlignment="Center"
+				Orientation="Horizontal"
+				Spacing="12">
+				<Image
+					Height="16"
+					VerticalAlignment="Center"
+					Source="{x:Bind ShellViewModel.FolderThumbnailImageSource, Mode=OneWay}" />
+				<TextBlock
+					VerticalAlignment="Center"
+					Style="{StaticResource BodyStrongTextBlockStyle}"
+					Text="{helpers:ResourceString Name=FilteringFor}" />
+				<AutoSuggestBox
+					x:Name="FilterTextBox"
+					Width="180"
+					VerticalAlignment="Center"
+					PlaceholderText="{helpers:ResourceString Name=Filename}"
+					PreviewKeyDown="FilterTextBox_PreviewKeyDown"
+					TextChanged="FilterTextBox_TextChanged" />
```

Comment:
Note that using TwoWay binding prevents the folder content from loading when refreshing or navigating to a different location.